pandas/core/indexes/interval.py

```diff
@@ -461,7 +461,24 @@ def is_unique(self):
         """
         Return True if the IntervalIndex contains unique elements, else False
         """
-        return self._multiindex.is_unique
+        left = self.left
+        right = self.right
+
+        if self.isna().sum() > 1:
+            return False
+
+        if left.is_unique or right.is_unique:
+            return True
+
+        seen_pairs = set()
+        check_idx = np.where(left.duplicated(keep=False))[0]
```

Comment:
acutally can't you just do this

```
pairs = [(left[idx], right[idx] for idx in checks_idx]
return len(set(pairs)) == len(pairs)
```

Reply:
The present approach may be better in which `False` is returned once a duplicate is found.
To compare the length, we run over all potential duplicates.